src/Otp.php

```diff
@@ -91,9 +91,20 @@ public function totp($secret, $timecounter = null)
     /* (non-PHPdoc)
      * @see Otp.OtpInterface::checkHotp()
     */
-    public function checkHotp($secret, $counter, $key)
+    public function checkHotp($secret, $counter, $key, $counterwindow = 5)
```

Comment:
How about something like this?

```
 protected $hotp_counter;

// some code 

 public function checkHotp($secret, $counter, $key){
  // old code
 }

public function checkHotpWithResync($secret, $counter, $key, $counterwindow = 2){
  if(!is_numeric($counterwindow) || $counterwindow < 0){
          throw new \InvalidArgumentException('Invalid counterwindow supplied');
  }
  for($c = 0; $c <= $counterwindow; $c = $c + 1) {
          if($this->safeCompare($this->hotp($secret, $counter + $c), $key)){
              $this->hotp_counter = $counter+$c;
              return true;
          }
  }
 return false;
}

public function getHotpCounter(){
   return $this->$hotp_counter;
}
```
This should preserve backward compatiblity and prevent the counter from being leaked.
As for the security issue, the system should block the code when it gets a successful validation. It would block the key which resulted in the successful validation and not have anything to do with the counter.